File: 0001-two-sum/0001-two-sum.cpp
#include <iostream>
#include <vector>
class Solution {
public:
  std::vector<int> twoSum(std::vector<int> &nums, int target) {
    std::vector<int> ans;
    for (int i = 0; i < nums.size(); i++) {
      for (int j = i + 1; j < nums.size(); j++) {
        if (nums[i] + nums[j] == target) {
          ans.push_back(i);
          ans.push_back(j);
          return ans;
        }
      }
    }
    return ans;
  }
};

int main() {
  Solution solution;
  std::vector<int> nums1 = {2, 7, 11, 15};
  int target = 26;
  std::vector<int> example1 = solution.twoSum(nums1, target);
  std::cout << "example1: ";
  for (int i = 0; i < example1.size(); i++) {
    std::cout << example1[i] << " ";
  }
  std::cout << std::endl;
  if (example1 == std::vector<int>{0, 1}) {
    std::cout << "good " << std::endl;
  } else {
    std::cout << "not good" << std::endl;
  }
}

File: 0001-two-sum/0001-two-sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}
        for (idx, num) in enumerate(nums):
            if num in hashmap:
                return [hashmap[num], idx]
            res = target - num
            hashmap[res] = idx




            
            

solution = Solution()

def testTwoSum():
    test_cases = [
        {
            "input": {
                "nums": [2,7,11,15],
                "target": 9
            },
            "output": [0, 1]
        },
        {
            "input": {
                "nums": [3,2,4],
                "target": 6
            },
            "output": [1, 2]
        },
        {
            "input": {
                "nums": [3,3],
                "target": 6
            },
            "output": [0, 1]
        }
    ]

    for idx, test in enumerate(test_cases):
        result = solution.twoSum(**test["input"])
        try:
            assert result.sort() == test["output"].sort(), f"Test {idx + 1} failed. Expected {test['output']}, but got {result}"
            print(f"Test {idx + 1} passed! ✅")
        except Exception as e:
            print(f"Test {idx + 1} failed with error: {e} ❌")


if __name__ == "__main__":
    testTwoSum()
